core/types/receipt.go

```diff
@@ -27,89 +26,116 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
+// Receipt represents the results of a transaction.
 type Receipt struct {
+	// Consensus fields
 	PostState         []byte
 	CumulativeGasUsed *big.Int
 	Bloom             Bloom
-	TxHash            common.Hash
-	ContractAddress   common.Address
-	logs              state.Logs
-	GasUsed           *big.Int
+	Logs              state.Logs
+
+	// Implementation fields
+	TxHash          common.Hash
+	ContractAddress common.Address
+	GasUsed         *big.Int
 }
 
+// NewReceipt creates a barebone transaction receipt, copying the init fields.
 func NewReceipt(root []byte, cumalativeGasUsed *big.Int) *Receipt {
```

Comment:
Not my typo, but will correct in a follow up PR that does a few more changes all over core :)